kubeconfig_plugin.py

```diff
@@ -117,57 +138,106 @@ class ErrorOutput:
-            except Exception:
-                print("token missing for user in kubeconfig")
-
-        if (
-            "client_key_data" not in output
-            and "client_certificate_data" not in output
-            and "token" not in output
-        ):
+        try:
+            context = None
+            for ctx in kubeconfig["contexts"]:
+                if ctx["name"] == current_context:
+                    context = ctx["context"]
+            if context is None:
+                return "error", ErrorOutput(
+                    "Failed to find a context named {} in the kubeconfig file.".format(current_context))
+            current_cluster = context["cluster"]
+            current_user = context["user"]
+        except Exception as e:
             return "error", ErrorOutput(
-                1, "Both client data and token missing, exiting!"
-            )
+                "Failed to find a context named {} in the kubeconfig file: {}".format(current_context, e.__str__()))
 
-        print(output)
+        try:
+            cluster = None
+            for cl in kubeconfig["clusters"]:
+                if cl["name"] == current_cluster:
+                    cluster = cl["cluster"]
+            if cluster is None:
+                return "error", ErrorOutput(
+                    "Failed to find a cluster named {} in the kubeconfig file.".format(current_cluster))
+        except Exception as e:
+            return "error", ErrorOutput(
+                "Failed to find a cluster named {} in the kubeconfig file: {}".format(current_cluster, e.__str__()))
+
+        try:
+            user = None
+            for u in kubeconfig["users"]:
+                if u["name"] == current_user:
+                    user = u["user"]
+            if user is None:
+                return "error", ErrorOutput(
+                    "Failed to find a user named {} in the kubeconfig file.".format(current_user))
+        except Exception as e:
+            return "error", ErrorOutput(
+                "Failed to find a user named {} in the kubeconfig file: {}".format(current_user, e.__str__()))
 
-        return "success", kubeconfig_output_schema.unserialize(output)
-    except Exception:
-        return "error", ErrorOutput(1, "Failure in parsing kubeconfig:")
+        output = SuccessOutput(
+            Connection(cluster["server"]),
+            None,
+        )
+        try:
+            output.connection.cacert = base64.b64decode(cluster[
+                                                            "certificate-authority-data"
+                                                        ]).decode('ascii')
+        except KeyError:
+            pass
+        try:
+            output.connection.cert = base64.b64decode(user[
+                                                          "client-certificate-data"
+                                                      ]).decode('ascii')
+        except KeyError:
+            pass
```

Comment:
Could you wrap thos in a dedicated function? 
```python
def getFromDict(d -> dict, key -> str) -> str:
   try:
      return d[key]
   except KeyError:
      pass
```

Reply:
Why can one use:
```
d.get(key, "")
```

Reply:
The format that the schema needs is `None` for all missing optional values. We can't use get for this, so the only ways for this to work is the current way, or a just-as-verbose `in` conditional.